async-gateway/service.go

```diff
@@ -127,6 +119,38 @@ func (s *service) GetWorkload(id string) (GetWorkloadResponse, error) {
 	}, nil
 }
 
+func (s *service) getStatus(id string) (Status, error) {
+	prefix := s.workloadStoragePrefix()
+	log := s.logger.With(zap.String("id", id))
+
+	// download workload status
+	log.Debug("checking status", zap.String("path", fmt.Sprintf("%s/%s/*", prefix, id)))
+	files, err := s.storage.List(fmt.Sprintf("%s/%s", prefix, id))
+	if err != nil {
+		return Status(""), err
```

Comment:
nit: return Status{}, err

Reply:
I think that would be an invalid syntax. It's not a struct.

Reply:
right, in that case: 
return "", err